### DISCUSSION TITLE
why the task channel of the workers not be closed

### DISCUSSION THREAD

**Discussion by yugu2day** [panjf2000/ants#341] [Q&A] [answered]
I found that when purging/resetting the workers, simply setting it to nil and sending nil to the task channel so that it can stop the for range loop in the worker's run function. 
```
func (wq *workerStack) reset() {
	for i := 0; i < wq.len(); i++ {
		wq.items[i].finish()
		wq.items[i] = nil
	}
	wq.items = wq.items[:0]
}

```

However, the task channel remains open although it can be garbage collected finally. But why don't we close it directly?

**Comment by panjf2000** [comment, marked as answer]
Like you said, there wouldn't be any memory leaks, so why should we bother to close it? I don't see any obligated instructions about closing the used channels in the [Go specification](https://go.dev/ref/spec#Channel_types) and [effective Go](https://go.dev/doc/effective_go#channels). Generally speaking, we need a reason to do something more than a reason not to do something.

**Reply by yugu2day** [reply]
got it, thanks for your explanation.
